pkg/check/interface.go

```diff
@@ -14,6 +16,40 @@ import (
 	"k8s.io/legacy-cloud-providers/vsphere"
 )
 
+// TagManager defines an interface to an implementation of the AuthorizationManager to facilitate mocking.
+type TagManager interface {
+	AttachTag(ctx context.Context, tagID string, ref mo.Reference) error
+	CreateCategory(ctx context.Context, category *vapitags.Category) (string, error)
+	CreateTag(ctx context.Context, tag *vapitags.Tag) (string, error)
+	DeleteCategory(ctx context.Context, category *vapitags.Category) error
+	DeleteTag(ctx context.Context, tag *vapitags.Tag) error
+	DetachTag(ctx context.Context, tagID string, ref mo.Reference) error
+	ListCategories(ctx context.Context) ([]string, error)
+	ListTags(ctx context.Context) ([]string, error)
+	GetCategories(ctx context.Context) ([]vapitags.Category, error)
+	GetCategory(ctx context.Context, id string) (*vapitags.Category, error)
+	GetTagsForCategory(ctx context.Context, id string) ([]vapitags.Tag, error)
+	GetAttachedTags(ctx context.Context, ref mo.Reference) ([]vapitags.Tag, error)
+	GetAttachedTagsOnObjects(ctx context.Context, objectID []mo.Reference) ([]vapitags.AttachedTags, error)
+	GetAttachedObjectsOnTags(ctx context.Context, tagID []string) ([]vapitags.AttachedObjects, error)
```

Comment:
same for this. Is this interface actually necessary? Are we creating mock concrete implementations that use this interface so as code can be tested?